modin/engines/ray/cudf_on_ray/frame/gpu_manager.py

```diff
@@ -0,0 +1,76 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import ray
+import cudf
+import pandas
+from .partition import cuDFOnRayFramePartition
+
+
+@ray.remote(num_gpus=1)
+class GPUManager(object):
+    def __init__(self, gpu_id):
+        self.key = 0
+        self.cudf_dataframe_dict = {}
+        self.gpu_id = gpu_id
+
+    def apply_non_persistent(self, first, other, func, **kwargs):
```

Comment:
Why is this different from `apply`? Some method level documentation here would be quite helpful.

Reply:
`apply` is meant to be used when you want to store the dataframe into the GPU cache. For example, `new_df = df.mean()` where `apply_not_persistent` is used when intermediate computation within the GPU manager do not want to be stored. For example, within a broadcast join you do not want to store the intermediate join results per each partition but just the final concatenation of all of them. Therefore, you only want to store a specific value but you are using the GPU to speed up the computation without taking space out of it. I agree with you level documentation will make this better.